### DISCUSSION TITLE
What about adding unit tests ?

### DISCUSSION THREAD

**Discussion by marques-kevin** [heybereket/oasis#40] [Ideas]
As the project will grow, more we will need a unit-test to ensure that the application still working. But adding test as a cost in many sides : 

- We need to refactor the project to be testable (for example, separate the components from the API calls (Separation of Concerns)
- Each new features should be tested so the development process will be longer
- It will be harder for contributors to enter in the project and make easy updates

**Comment by marques-kevin** [comment]
This issue is a good example of why we need tests #43

**Reply by HenryLeC** [reply]
Unit tests have been added to the api and we have cypress setup for E2E testing.
